src/webgpu/shader/execution/builtin/isnan.spec.ts

```diff
@@ -0,0 +1,110 @@
+export const description = `WGSL execution test. Section: Value-testing built-in functions. Function: isNan`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { assert } from '../../../../common/util/util.js';
+import { GPUTest } from '../../../gpu_test.js';
+import { NumberRepr } from '../../../util/conversion.js';
+import { generateTypes } from '../../types.js';
+import { subnormalF32Examples, normalF32Examples } from '../../values.js';
+
+import { runValueCheckTest } from './value_testing_built_in_functions.spec.js';
+
+export const g = makeTestGroup(GPUTest);
+
+g.test('value_testing_builtin_functions,isNan')
+  .uniqueId('fdd1e7105af70b74')
+  .specURL('https://www.w3.org/TR/2021/WD-WGSL-20210929/#value-testing-builtin-functions')
+  .desc(
+    `
+isNan:
+I is f32 or vecN<f32> T is bool if I is a scalar, or vecN<bool> if I is a vector isNan(e: I ) -> T Test for NaN according to IEEE-754. Component-wise when I is a vector. (OpIsNan)
+
+Please read the following guidelines before contributing:
+https://github.com/gpuweb/cts/blob/main/docs/plan_autogen.md
+`
+  )
+  .params(u =>
+    u
+      .combine('baseType', ['f32'] as const)
+      .combine('containerType', ['scalar', 'vector'] as const)
+      .combine('storageClass', ['storage'] as const) // Needed by generateTypes
+      .expandWithParams(generateTypes)
+  )
+  .fn(async t => {
+    assert(t.params._kTypeInfo !== undefined, 'generated type is undefined');
+
+    // Run cases using JavaScript floating point values and comparisons.
+    const make = (the_input: number, expected: boolean) => {
+      return {
+        input: NumberRepr.fromF32(the_input),
+        expected: [NumberRepr.fromF32(expected ? 1.0 : 0.0)],
+      };
+    };
+
+    const casesFromFloats = [
+      // Finite
+      make(0.0, false),
+      make(10.0, false),
+      make(-10.0, false),
+      // Infinities
+      make(Infinity, false),
+      make(-Infinity, false),
+      // NaNs
+      make(NaN, true),
+      make(-NaN, true),
+    ]
+      // Normals are not Nan
+      .concat(normalF32Examples().map(n => make(n, false)))
+      // Subnormals are not Nan
+      .concat(subnormalF32Examples().map(n => make(n, false)));
+
+    runValueCheckTest(
+      t,
+      t.params.type,
+      t.params._kTypeInfo.arrayLength,
+      'isNan',
+      Float32Array,
+      casesFromFloats
+    );
+
+    // Run exotic cases using explicit bit Uint representations of floats.
+    const makeUint = (bits: number, expected: boolean) => {
+      const expected_bits = NumberRepr.fromF32(expected ? 1.0 : 0).bits[0];
+      return {
+        input: NumberRepr.fromU32(bits),
+        expected: [NumberRepr.fromU32(expected_bits)],
+      };
+    };
+
+    const casesFromUints = [
+      makeUint(0x7f80_0000, false), // infinity
+      makeUint(0xff80_0000, false), // -infinity
```

Comment:
I'd suggest using the constants form https://github.com/gpuweb/cts/blob/2961e70ac940aa861f924efd1ec97405a0afc4e5/src/webgpu/shader/execution/builtin/builtin.ts#L211.